python/ray/__init__.py

```diff
@@ -11,9 +11,15 @@ def _configure_system():
 
     """Wraps system configuration to avoid 'leaking' variables into ray."""
 
-    # MUST add pickle5 to the import path because it will be imported by some
-    # raylet modules.
+    # Sanity check pickle5 if it has been installed.
     if "pickle5" in sys.modules:
+        if sys.version_info >= (3, 8):
```

Comment:
I still think we need a warning - otherwise users are unable to use Ray if another dependency ship pickle5. Even with pickle5, workarounds are possible. If we raise import error, users would give up using Ray.

Reply:
Agree. But what about the exceptions below? They are also non-critical issues that do not affect Ray itself.

Reply:
I think it is both ok to keep or remove the exception.

Reply:
Sounds good! I will use warning for all pickle5 checks, since failing these checks may not prevent Ray from running correctly.